app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java

```diff
@@ -1731,6 +1732,12 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                 }
             }
 
+            if (getBestCollectorHardwareName().equals("G7")) { // Remove battery options, from G5/G6/Dex1/G7 Debug Settings, if we are using G7 or One+
```

Comment:
Really good idea to wrap this with a try{} catch{} pair and log at wtf level if an exception is generated, otherwise future changes to preferences structure can cause crashes.

Reply:
Sure, thanks.  It will be done.

Reply:
Done